src/Aspire.Hosting.PostgreSQL/PostgresBuilderExtensions.cs

```diff
@@ -162,39 +161,21 @@ public static IResourceBuilder<T> WithPgAdmin<T>(this IResourceBuilder<T> builde
 
             builder.ApplicationBuilder.Eventing.Subscribe<AfterEndpointsAllocatedEvent>((e, ct) =>
             {
-                // Add the servers.json file bind mount to the pgAdmin container
-
                 var postgresInstances = builder.ApplicationBuilder.Resources.OfType<PostgresServerResource>();
 
-                // Create servers.json file content in a temporary file
-
-                var tempConfigFile = WritePgAdminServerJson(postgresInstances);
-
-                try
-                {
-                    var aspireStore = e.Services.GetRequiredService<IAspireStore>();
-
-                    // Deterministic file path for the configuration file based on its content
-                    var configJsonPath = aspireStore.GetFileNameWithContent($"{builder.Resource.Name}-servers.json", tempConfigFile);
-
-                    // Need to grant read access to the config file on unix like systems.
-                    if (!OperatingSystem.IsWindows())
-                    {
-                        File.SetUnixFileMode(configJsonPath, FileMode644);
-                    }
+                // Generate the contents of the servers.json file
+                var serversConfigFile = WritePgAdminServerJson(postgresInstances);
```

Comment:
Variable name is confusing, now it's a file content, not a "File".
We can assume these files are fine in memory, but I kind of used a physical file on purpose before.

Reply:
This API is specifically intended to replace using physical files in situations where we need to control the ownership and permissions. `docker cp` specifically doesn't create any kind of link back to the host filesystem so bind mounts are still appropriate when wanting to work with physical files from the host filesystem.